apps/web/src/stores/command-palette-store.ts

```diff
@@ -0,0 +1,207 @@
+/*
+This file is part of the Notesnook project (https://notesnook.com/)
+
+Copyright (C) 2023 Streetwriters (Private) Limited
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import BaseStore from ".";
+import { db } from "../common/db";
+import createStore from "../common/store";
+import {
+  Icon,
+  Note as NoteIcon,
+  Notebook as NotebookIcon,
+  Reminder as ReminderIcon,
+  Tag as TagIcon
+} from "../components/icons";
+import { commands as COMMANDS } from "../dialogs/command-palette/commands";
+import { hashNavigate, navigate } from "../navigation";
+import { useEditorStore } from "./editor-store";
+
+export interface Command {
+  id: string;
+  title: string;
+  type: "command" | "note" | "notebook" | "tag" | "reminder";
+  group: string;
+}
+
+const CommandIconMap = COMMANDS.reduce((acc, command) => {
+  acc.set(command.id, command.icon);
+  return acc;
+}, new Map<string, Icon>());
+
+const CommandActionMap = COMMANDS.reduce((acc, command) => {
+  acc.set(command.id, command.action);
+  return acc;
+}, new Map<string, (arg?: any) => void>());
+
+const CommandTypeItems = COMMANDS.map((c) => ({
+  id: c.id,
+  title: c.title,
+  group: c.group,
+  type: "command" as const
+}));
+
+class CommandPaletteStore extends BaseStore<CommandPaletteStore> {
+  commands: Command[] = CommandTypeItems;
+  selected = 0;
+  query = ">";
+
+  setCommands = (commands: Command[]) => {
+    this.set((state) => {
+      state.commands = commands;
+    });
+  };
+
+  setSelected = (selected: number) => {
+    this.set((state) => {
+      state.selected = selected;
+    });
+  };
+
+  setQuery = (query: string) => {
+    this.set((state) => {
+      state.query = query;
+    });
+  };
```

Comment:
Why are we storing the query in the store? Is it actually needed anywhere else? Can this be in local state instead of in a zustand store? Same for other properties in this store.

Reply:
I first made the store to move all logic outside of the component. After these review comments, I've realized we can make do with local state and put all other functions in a static util class